app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,12 @@
+def create
+  @post = Post.find(params[:post_id])
+  @like = Like.new(user: current_user, post: @post)
+
+  if @like.save
+    flash[:success] = 'Liked successfully!'
+    redirect_to user_post_path(@post.user, @post)
+  else
+    flash[:alert] = 'Something went wrong'
+    render :new
+  end
+end
```

Comment:
- [ ] Since the above bugs prevented the creation of a post I was also unable to provide a like to the posts so please make sure there's no bug on that end as well.